geoips/geoips_utils.py

```diff
@@ -423,3 +423,49 @@ def is_editable(package_name):
             return True
     # Package is installed in non-editable mode
     return False
+
+
+def get_source_name(xobjs):
+    """Retrieve the source name from the metadata provided in xobjs.
+
+    Where xobjs is a dictionary of xarray objects which contain only metadata for the
+    data which will be processed.
+
+    Parameters
+    ----------
+    xobjs: dict of xarray Datasets
+        - Dictionary of xarray Datasets which contain metadata about the data currently
+          being used in processing.
+
+    Returns
+    -------
+    source_name: str
+        - The source name in which the data is coming from
+    """
+    if hasattr(xobjs["METADATA"], "source_name"):
+        # This means metadata is coming from a single file
+        source_name = xobjs["METADATA"].source_name
+    elif hasattr(
+        xobjs["METADATA"]
+        .get(
+            "source_file_attributes",
+            {},
+        )
+        .get("filename_0"),
+        "source_name",
+    ):
+        # If the metadata returned has the format:
+        # xobjs["METADATA"]["source_file_attributes"]["filename_0"] ...
```

Comment:
I don't think this should be a hard coded "filename_0".  I believe it is intended to be the actual filename whose attributes you are enumerating. This can either be a base path (excluding path on disk), or the full path with GeoIPS env variables replacing portions of the full path (ie, GEOIPS_OUTDIRS, GEOIPS_TESTDATA_DIR, etc.  There is a function in geoips_utils.py that replacing GeoIPS env vars in a path, you can use that if you want your filenames to include the full path).

As explained in the xarray standards:
https://github.com/NRLMMD-GEOIPS/geoips/blob/main/docs/source/devguide/xarray_standards.rst#xarray-standard-attributes

Reply:
Could you please leave a link to the commit(s) where this was resolved?  I went to quickly look for it, and it was not obvious which commit made this update.

Reply:
Maybe these, are there others?
* https://github.com/NRLMMD-GEOIPS/geoips/pull/427/commits/d4eeb2855ef29953f07f51718f39d6326079c443
* https://github.com/NRLMMD-GEOIPS/geoips/pull/427/commits/7ce0ecb8ab2382898f47a885fcb3b6dc3389ba39

Reply:
@evrose54 Provide links to show where this is resolved.

Reply:
[Here](https://github.com/NRLMMD-GEOIPS/geoips/pull/427/files#r1821411519) is where it was resolved.

Reply:
Thanks!